### PR TITLE
feat(system-e2e): Move system-e2e to NX, v4

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
Continued from broken #14505 

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced Playwright testing enhancements across the system, including updated configurations and test examples.
  - Updated command-line instructions for system end-to-end testing to enhance usability.
- **Documentation**
  - Revised documentation for system end-to-end tests to reflect new command options and configurations.
- **Refactor**
  - Standardized logging across various support files from `debug` to `logger` for consistency and clarity in debugging outputs.
- **Bug Fixes**
  - Enabled previously skipped test cases in the Admin portal, ensuring comprehensive test coverage.
- **Chores**
  - Updated `.gitignore` to exclude specific directories and files associated with Playwright testing, ensuring cleaner version control.

### CODE REVIEW

**Review by saevarma — APPROVED** [review]
Core files LGTM

File: apps/system-e2e/.eslintrc.json [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+    {
+      "files": ["*.ts", "*.tsx"],
+      "rules": {}
+    },
+    {
+      "files": ["*.js", "*.jsx"],
+      "rules": {}
+    },
+    {
+      "files": ["e2e/**/*.{ts,js,tsx,jsx}"],
```
Comment (lodmfjord) [inline apps/system-e2e/.eslintrc.json:19]: Why is this extra override needed?

File: apps/system-e2e/entrypoint.sh [hunk 2]
```diff
@@ -14,24 +14,28 @@ if [[ "$*" =~ --project ]]; then
 fi
 
 export TEST_PROJECT TEST_ENVIRONMENT TEST_TYPE TEST_RESULTS_S3 TEST_FILTER
-
-DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+export PATH="./node_modules/.bin:$PATH"
```
Comment (lodmfjord) [inline apps/system-e2e/entrypoint.sh:17]: Relative path in PATH?

File: apps/system-e2e/src/support/utils.ts [hunk 3]
```diff
@@ -1,30 +1,20 @@
...
 import fs from 'fs'
+import { getLogger } from '../../../../infra/src/logging'
 
 export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
+export const logger = getLogger()
+export const debug = logger.debug
 
-export function createMockPdf() {
+export const createMockPdf = () => {
   fs.writeFile('./mockPdf.pdf', 'test', (e) => {
```
Comment (lodmfjord) [inline apps/system-e2e/src/support/utils.ts:10]: This creates a non-async function that calls a function with a hook and returns before file is created. Can this create an error?

File: apps/system-e2e/src/support/utils.ts [hunk 4]
```diff
@@ -1,30 +1,20 @@
...
-  const doc = new PDFDocument()
-  doc.text('test')
-  doc.save()
-  doc.pipe(fs.createWriteStream('./mockPdf.pdf'))
-  doc.end()
-  */
 }
 
-export function deleteMockPdf() {
+export const deleteMockPdf = () => {
```
Comment (lodmfjord) [inline apps/system-e2e/src/support/utils.ts:15]: Same thing…

File: apps/system-e2e/src/tests/islandis/admin-portal/smoke/tenant-application.spec.ts [hunk 5]
```diff
@@ -359,7 +363,7 @@ test.describe('Admin portal application', () => {
         await expect(page.getByRole('heading', { name: title })).toBeVisible()
         await Promise.all(
           [...checkboxes, accessTokenLifetime, customClaims].map((checkbox) =>
-            expect(checkbox).not.toBeVisible(),
+            expect(checkbox).toBeHidden(),
```
Comment (lodmfjord) [inline apps/system-e2e/src/tests/islandis/admin-portal/smoke/tenant-application.spec.ts:366]: toBeHidden also is true if element is not rendered. Is this expected behavior?

**Review by lodmfjord — COMMENTED** [review]
Minor comments

File: apps/system-e2e/.eslintrc.json (hunk 1 above)
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/.eslintrc.json:19]: :shrug: Copy-pasted or generated

File: apps/system-e2e/entrypoint.sh (hunk 2 above)
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/entrypoint.sh:17]: Yes, so we _always_ get the local `node_modules` first.

File: apps/system-e2e/src/support/utils.ts (hunk 3 above)
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/src/support/utils.ts:10]: Probably. It's being used in tests, and they aren't failing :shrug: 

This should be revisited when lib-ified

**Review by lodmfjord — APPROVED** [review]
LGTM

**Pull request merged**